### PR TITLE
[Fluent] Radio Button

### PR DESCRIPTION
**Pull request by PaskalevStoyan**
Related to: https://github.com/telerik/kendo-themes-private/issues/3

### CODE REVIEW

File: packages/fluent/scss/radio/_layout.scss [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+@mixin kendo-radio--layout() {
+
+    .k-radio {
+        @include border-radius( $kendo-radio-radius );
+        width: var( --INTERNAL-kendo-radio-width, #{ $kendo-radio-width } );
+        height: var( --INTERNAL-kendo-radio-height, #{ $kendo-radio-height } );
+        margin: 0;
+        padding: 0;
+        line-height: initial;
+        border-width: $kendo-radio-border-width;
```
Comment (Juveniel) [inline packages/fluent/scss/radio/_layout.scss:15]: use css var here

File: packages/fluent/scss/radio/_layout.scss [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+        &::before {
+            font-size: var( --INTERNAL-kendo-radio-font-size, #{ $kendo-radio-font-size } )
+        }
+    }
+
+
+    // Radio indicator
+    .k-radio::before {
+        @if $kendo-radio-indicator-type == "glyph" {
+            content: $kendo-radio-checked-glyph;
```
Comment (Juveniel) [inline packages/fluent/scss/radio/_layout.scss:38]: use css var here

File: packages/fluent/scss/radio/_layout.scss [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+            width: 1em;
+            height: 1em;
+            font-family: $kendo-radio-glyph-font-family;
```
Comment (Juveniel) [inline packages/fluent/scss/radio/_layout.scss:41]: use css var here

File: packages/fluent/scss/radio/_layout.scss [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+        &:not(
+            .k-radio.k-invalid, .k-radio:invalid,
+            .k-radio.k-disabled, .k-radio:disabled
+        )::after {
+            content: "";
+            width: var( --INTERNAL-kendo-radio-checked-width, #{ $kendo-radio-checked-width });
+            height: var( --INTERNAL-kendo-radio-checked-height, #{ $kendo-radio-checked-height });
+            border-width: var( --INTERNAL-kendo-radio-checked-border-width, #{ $kendo-radio-checked-border-width });
+            border-style: solid;
+            border-radius: var( --kendo-radio-radius, #{ $kendo-radio-radius });
```
Comment (Juveniel) [inline packages/fluent/scss/radio/_layout.scss:81]: name variables `kendo-radio-border-radius`

File: packages/fluent/scss/radio/_layout.scss [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+    }
+
+
+    // Radio label
+    .k-radio-label {
+        margin: 0;
+        padding: 0;
+        display: inline-flex;
+        align-items: flex-start;
+        gap: $kendo-radio-label-margin-x;
```
Comment (Juveniel) [inline packages/fluent/scss/radio/_layout.scss:132]: use css var

File: packages/fluent/scss/radio/_layout.scss [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+            // Hide ripple temporarily
+            visibility: hidden !important; // sass-lint:disable-line no-important
+        }
+    }
+    .k-radio + .k-label,
+    .k-radio-wrap + .k-label,
+    .k-radio + .k-radio-label,
+    .k-radio-wrap + .k-radio-label {
+        display: inline;
+        margin-inline-start: $kendo-radio-label-margin-x;
```
Comment (Juveniel) [inline packages/fluent/scss/radio/_layout.scss:147]: use css var

File: packages/fluent/scss/radio/_layout.scss [hunk 7]
```diff
@@ -0,0 +1,261 @@
...
+        display: flex;
+        flex-flow: column nowrap;
+        gap: 0;
+        list-style: none;
+    }
+
+
+    .k-radio-item,
+    .k-radio-list-item {
+        padding: $kendo-radio-list-item-padding-y $kendo-radio-list-item-padding-x;
```
Comment (Juveniel) [inline packages/fluent/scss/radio/_layout.scss:173]: convert to `padding-inline` and `padding-block`

File: packages/fluent/scss/radio/_layout.scss [hunk 8]
```diff
@@ -0,0 +1,261 @@
...
+        display: flex;
+        flex-flow: row nowrap;
+        align-items: center;
+        align-content: center;
+        gap: map-get( $kendo-spacing, 1 );
```
Comment (Juveniel) [inline packages/fluent/scss/radio/_layout.scss:178]: extract var

File: packages/fluent/scss/radio/_layout.scss [hunk 9]
```diff
@@ -0,0 +1,261 @@
...
+            margin: 0;
+        }
+    }
+
+
+    .k-radio-list-horizontal,
+    .k-radio-list.k-list-horizontal {
+        display: flex;
+        flex-flow: row wrap;
+        gap: $kendo-radio-list-spacing;
```
Comment (Juveniel) [inline packages/fluent/scss/radio/_layout.scss:190]: use css var

File: packages/fluent/scss/radio/_layout.scss [hunk 10]
```diff
@@ -0,0 +1,261 @@
...
+                --INTERNAL-kendo-radio-font-size: #{ $_glyph-size };
+            }
+        }
+        
+        .k-radio-#{$size}.k-checked,
+        .k-radio-#{$size}:checked,
+        .k-radio-#{$size}.k-hover,
+        .k-radio-#{$size}:hover {
+            &::after {
+
```
Comment (Juveniel) [inline packages/fluent/scss/radio/_layout.scss:245]: remove empty line

File: packages/fluent/scss/radio/_layout.scss [hunk 11]
```diff
@@ -0,0 +1,261 @@
...
+            }
+        }
+
+        .k-ripple-container {
+            .k-radio-#{$size}::after {
+                --INTERNAL-kendo-radio-ripple-width: #{ $_ripple-size };
+                --INTERNAL-kendo-radio-ripple-height: #{ $_ripple-size };
+            }
+        }
+    }
```
Comment (Juveniel) [inline packages/fluent/scss/radio/_layout.scss:260]: add empty line

File: packages/fluent/scss/radio/_variables.scss [hunk 12]
```diff
@@ -0,0 +1,198 @@
...
+
+/// Default width of radio button.
+/// @group radio
+$kendo-radio-width: map-get( $kendo-spacing, 5) !default;
+/// Default height of radio button.
+/// @group radio
+$kendo-radio-height: map-get( $kendo-spacing, 5) !default;
+/// Default font size of radio button.
+/// @group radio
+$kendo-radio-font-size: ( map-get( $kendo-spacing, 5 ) - map-get( $kendo-spacing, thin ) ) !default;
```
Comment (Juveniel) [inline packages/fluent/scss/radio/_variables.scss:49]: can we use --font-size-lg here ?

File: packages/fluent/scss/radio/_variables.scss [hunk 13]
```diff
@@ -0,0 +1,198 @@
...
+/// Default ripple width of radio button.
+/// @group radio
+$kendo-radio-ripple-width: map-get( $kendo-spacing, 5 ) * 3 !default;
```
Comment (Juveniel) [inline packages/fluent/scss/radio/_variables.scss:52]: calc( $kendo-radio-width * 3 ) ?

File: packages/fluent/scss/radio/_variables.scss [hunk 14]
```diff
@@ -0,0 +1,198 @@
...
+/// Default ripple height of radio button.
+/// @group radio
+$kendo-radio-ripple-height: map-get( $kendo-spacing, 5 ) * 3 !default;
```
Comment (Juveniel) [inline packages/fluent/scss/radio/_variables.scss:55]: calc( $kendo-radio-height * 3 ) ?

**Review by Juveniel — CHANGES_REQUESTED** [review]

File: packages/fluent/scss/radio/_variables.scss [hunk 15]
```diff
@@ -0,0 +1,198 @@
...
+
+/// Background color of hovered radio button.
+/// @group radio
+$kendo-radio-hover-bg: $kendo-checkbox-hover-bg !default;
+/// Color of hovered radio button.
+/// @group radio
+$kendo-radio-hover-text: $kendo-checkbox-hover-text !default;
+/// Border color of hovered radio button.
+/// @group radio
+$kendo-radio-hover-border: get-theme-color(neutral, 130) !default;
```
Comment (epetrow) [inline packages/fluent/scss/radio/_variables.scss:75]: Isn't it better to use get-theme-color-var() here?

File: packages/fluent/scss/radio/_variables.scss [hunk 16]
```diff
@@ -0,0 +1,198 @@
...
+
+/// Border of hovered and checked radio.
+/// @group checkbox
+$kendo-radio-checked-hover-border: get-theme-color( primary, 150 ) !default;
```
Comment (epetrow) [inline packages/fluent/scss/radio/_variables.scss:79]: Use get-theme-color-var() here as well

File: packages/fluent/scss/radio/_variables.scss [hunk 17]
```diff
@@ -0,0 +1,198 @@
...
+
+/// Border color of focused radio button.
+/// @group radio
+$kendo-radio-focus-border: $kendo-checkbox-focus-border !default;
+/// Box shadow of focused radio button.
+/// @group radio
+$kendo-radio-focus-shadow: $kendo-checkbox-focus-shadow !default;
+/// Border color of focused and checked radio button.
+/// @group radio
+$kendo-radio-focus-checked-border: get-theme-color(neutral, 130) !default;
```
Comment (epetrow) [inline packages/fluent/scss/radio/_variables.scss:114]: Use get-theme-color-var() and add spacing for consistency

File: packages/fluent/scss/radio/_variables.scss (hunk 12 above)
Comment (PaskalevStoyan) [reply packages/fluent/scss/radio/_variables.scss:49]: The `--font-size-lg` is 16px but the default font-size is 2 pixels lower than the size of the radio (which is 20px by default) thus the calculation is needed and we cannot use the font-size-lg variable.

File: packages/fluent/scss/radio/_layout.scss [hunk 18]
```diff
@@ -0,0 +1,262 @@
...
+
+
+    // Radio label
+    .k-radio-label {
+        margin: 0;
+        padding: 0;
+        display: inline-flex;
+        align-items: flex-start;
+        gap: var( --kendo-radio-label-margin-x, #{ $kendo-radio-label-margin-x } );
```
Comment (Juveniel) [inline packages/fluent/scss/radio/_layout.scss:132]: better name this `$kendo-radio-label-spacing`

File: packages/fluent/scss/radio/_variables.scss [hunk 19]
```diff
@@ -0,0 +1,200 @@
...
+
+/// Background color of hovered radio button.
+/// @group radio
+$kendo-radio-hover-bg: $kendo-checkbox-hover-bg !default;
+/// Color of hovered radio button.
+/// @group radio
+$kendo-radio-hover-text: $kendo-checkbox-hover-text !default;
+/// Border color of hovered radio button.
+/// @group radio
+$kendo-radio-hover-border: get-theme-color-var(neutral-130) !default;
```
Comment (Juveniel) [inline packages/fluent/scss/radio/_variables.scss:75]: add spacing

File: packages/fluent/scss/radio/_variables.scss [hunk 20]
```diff
@@ -0,0 +1,200 @@
...
+
+/// Top position of checked radio button.
+/// @group checked
+$kendo-radio-checked-top: map-get( $kendo-spacing, 1 ) !default;
+/// Left position of checked radio button.
+/// @group checked
+$kendo-radio-checked-left: map-get( $kendo-spacing, 1 ) !default;
+/// Border width of checked radio button.
+/// @group radio
+$kendo-radio-checked-border-width: ( map-get( $kendo-spacing, 1) + map-get( $kendo-spacing, hair)) !default;
```
Comment (Juveniel) [inline packages/fluent/scss/radio/_variables.scss:89]: add spacing

File: packages/fluent/scss/radio/_variables.scss [hunk 21]
```diff
@@ -0,0 +1,200 @@
...
+$kendo-radio-checked-width: calc( $kendo-radio-width / 2 ) !default;
+/// Height of checked radio button.
+/// @group radio
+$kendo-radio-checked-height: calc( $kendo-radio-height / 2 ) !default;
+/// Background color of checked radio button.
+/// @group radio
+$kendo-radio-checked-bg: var( --kendo-component-bg, initial) !default;
+/// Color of checked radio button.
+/// @group radio
+$kendo-radio-checked-text: get-theme-color-var(primary-130) !default;
```
Comment (Juveniel) [inline packages/fluent/scss/radio/_variables.scss:101]: spacing

File: packages/fluent/scss/radio/_variables.scss [hunk 22]
```diff
@@ -0,0 +1,200 @@
...
+
+/// Border color of focused radio button.
+/// @group radio
+$kendo-radio-focus-border: $kendo-checkbox-focus-border !default;
+/// Box shadow of focused radio button.
+/// @group radio
+$kendo-radio-focus-shadow: $kendo-checkbox-focus-shadow !default;
+/// Border color of focused and checked radio button.
+/// @group radio
+$kendo-radio-focus-checked-border: get-theme-color-var(neutral-130) !default;
```
Comment (Juveniel) [inline packages/fluent/scss/radio/_variables.scss:114]: spacing

File: packages/fluent/scss/radio/_variables.scss [hunk 23]
```diff
@@ -0,0 +1,200 @@
...
+
+/// Background color of disabled radio button.
+/// @group radio
+$kendo-radio-disabled-bg: $kendo-checkbox-disabled-bg !default;
+/// Color of disabled radio button.
+/// @group radio
+$kendo-radio-disabled-text: $kendo-checkbox-disabled-text !default;
+/// Border color of disabled radio button.
+/// @group radio
+$kendo-radio-disabled-border: get-theme-color(neutral, 60) !default;
```
Comment (Juveniel) [inline packages/fluent/scss/radio/_variables.scss:127]: spacing

**Review by Juveniel — CHANGES_REQUESTED** [review]

**Review by Juveniel — CHANGES_REQUESTED** [review]
I noticed additional things related to the design guidelines:

1) Hover state when item is not checked is not visible in the tests. Is this case covered ? If yes, we can add a test case for this.

2) When an item is checked and disabled, the inner dot is not visible. Refer to the design where this is showcased.

**Review by Juveniel — APPROVED** [review]

**Pull request merged**